### PR TITLE
[gitlab] Enhanced print-ci - ACIX-356

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

1. This adds an option (disabled by default) to expand `parallel: matrix: ...` fields for the gitlab ci. [Syntax documentation](https://docs.gitlab.com/ee/ci/yaml/#parallelmatrix)
2. It adds an option `--keep-special-objects` to keep non job objects such as `variables`, `stages` etc.

In addition, it renames the print task and moves it to the proper module.

### Motivation

Matrix configurations create jobs with different names (`new-e2e-containers: [--run TestKindSuite -c ddinfra:osDescriptor=ubuntu:22.04]` and `new-e2e-containers: [--run TestECSSuite]` have one gitlab ci config entry but have different names).
This option is useful to obtain the configuration for each of these created jobs.

### CODE REVIEW

File: tasks/libs/ciproviders/gitlab_api.py [hunk 1]
```diff
@@ -513,6 +517,60 @@ def filter_yaml(key, value):
...
+def expand_matrix_jobs(yml: dict) -> dict:
+    """
+    Will expand matrix jobs into multiple jobs
+    """
+    new_jobs = {}
+    to_remove = set()
+    for job in yml:
+        if 'parallel' in yml[job] and 'matrix' in yml[job]['parallel']:
+            to_remove.add(job)
+            for arg in yml[job]['parallel']['matrix']:
```
Comment (chouetz) [inline tasks/libs/ciproviders/gitlab_api.py:529]: - I would suggest this to be in a dedicated method with some tests
- I think your implementation is complex (but I maybe misunderstood something). Suggestion:
```
value_lists = []
for value in arg.values():
    if isinstance(value, str):
        value_lists.append([value])
    else:
        value_lists.append(value)
all_args = list(prod(*value_lists))
```
And to set variables to jobs you can zip the `arg.keys()` with each element of `all_args` (they are normally in the correct order)
wdyt?

File: tasks/libs/ciproviders/gitlab_api.py [hunk 2]
```diff
@@ -490,17 +491,20 @@ def flatten(yml):
     return flatten(yml)
 
 
-def filter_gitlab_ci_configuration(yml: dict, job: str | None = None) -> dict:
+def filter_gitlab_ci_configuration(yml: dict, job: str | None = None, keep_special_objects: bool = False) -> dict:
     """
     Filters gitlab-ci configuration jobs
 
     - job: If provided, retrieve only this job
+    - keep_special_objects: Will keep special objects (not jobs) in the configuration (variables, stages, etc.)
```
Comment (chouetz) [inline tasks/libs/ciproviders/gitlab_api.py:499]: why do we need to dismiss them?

Comment (CelianR) [reply tasks/libs/ciproviders/gitlab_api.py:499]: Some tasks process only jobs (e.g. the linter for `changes: paths: ...`) so these objects should not be always added

File: tasks/libs/ciproviders/gitlab_api.py [hunk 3]
```diff
@@ -513,6 +517,74 @@ def filter_yaml(key, value):
     return {node[0]: node[1] for node in (filter_yaml(k, v) for k, v in yml.items()) if node is not None}
 
 
+def expand_matrix_jobs(yml: dict) -> dict:
+    """
+    Will expand matrix jobs into multiple jobs
+    """
+
+    def get_combinated_variables(arg: dict[str, (str | list[str])]):
```
Comment (chouetz) [inline tasks/libs/ciproviders/gitlab_api.py:525]: 🥜 For readability I'm not fan of method declaration like this, I prefer it aside (and below the parent method)

**Review by chouetz — APPROVED** [review]

**Pull request merged**